axum/src/extract/raw_form.rs

```diff
@@ -0,0 +1,54 @@
+use async_trait::async_trait;
+use axum_core::extract::FromRequest;
+use http::{Method, Request};
+
+use super::{
+    has_content_type,
+    rejection::{InvalidFormContentType, RawFormRejection},
+};
+
+use crate::{body::HttpBody, BoxError};
+
+/// Extractor that extracts the raw form string, without parsing it.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use axum::{
+///     extract::RawForm,
+///     routing::get,
+///     Router
+/// };
+///
+/// async fn handler(RawForm(form): RawForm) {}
+///
+/// let app = Router::new().route("/", get(handler));
+/// # async {
+/// # axum::Server::bind(&"".parse().unwrap()).serve(app.into_make_service()).await.unwrap();
+/// # };
+/// ```
+#[derive(Debug)]
+pub struct RawForm(pub Option<String>);
```

Comment:
I think using an `Option` here might give the impression that it'll `None` instead of returning a rejection if something fails.

I'm wondering if we should just use an empty string in the case there is no query and remove the Option. What do you think?